src/components/application_manager/src/resumption/resumption_data_json.cc

```diff
@@ -148,32 +156,45 @@ void ResumptionDataJson::IncrementIgnOffCount() {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock autolock(resumption_lock_);
   Json::Value to_save = Json::arrayValue;
-  for (Json::Value::iterator it = GetSavedApplications().begin();
-       it != GetSavedApplications().end();
-       ++it) {
-    if ((*it).isMember(strings::ign_off_count)) {
-      Json::Value& ign_off_count = (*it)[strings::ign_off_count];
-      const uint32_t counter_value = ign_off_count.asUInt();
-      ign_off_count = counter_value + 1;
-    } else {
-      LOG4CXX_WARN(logger_, "Unknown key among saved applications");
-      Json::Value& ign_off_count = (*it)[strings::ign_off_count];
-      ign_off_count = 1;
+  Json::Value dictionary;
+  {
+    resumption::LastStateAccessor accessor =
+        last_state_wrapper_->get_accessor();
+    dictionary = accessor.GetData().dictionary();
+    Json::Value& saved_apps = GetSavedApplications(dictionary);
+    for (auto json_app = saved_apps.begin(); json_app != saved_apps.end();
+         ++json_app) {
+      if ((*json_app).isMember(strings::ign_off_count)) {
+        Json::Value& ign_off_count = (*json_app)[strings::ign_off_count];
+        const uint32_t counter_value = ign_off_count.asUInt();
+        ign_off_count = counter_value + 1;
+      } else {
+        LOG4CXX_WARN(logger_, "Unknown key among saved applications");
+        Json::Value& ign_off_count = (*json_app)[strings::ign_off_count];
+        ign_off_count = 1;
+      }
+      to_save.append(*json_app);
     }
-    to_save.append(*it);
+    SetSavedApplication(to_save, dictionary);
+    SetLastIgnOffTime(time(NULL), dictionary);
   }
-  SetSavedApplication(to_save);
-  SetLastIgnOffTime(time(NULL));
-  LOG4CXX_DEBUG(logger_, GetResumptionData().toStyledString());
+  {
+    resumption::LastStateAccessor accessor =
+        last_state_wrapper_->get_accessor();
```

Comment:
@LitvinenkoIra this one has no sense.
Why do you need to re-acquire accessor here as you already has another one?
Just use existing accessor:
```
    SetSavedApplication(to_save, dictionary);
    SetLastIgnOffTime(time(NULL), dictionary);
    accessor.GetMutableData().set_dictionary(dictionary);
```
P.S. you could remove scopes from line 160 as you need accessor to be alive till the end of this function

Reply:
@AKalinich-Luxoft fixed 585843bef0ad1e839a35048f1ee4eef77c9eaabc